org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterGenericButton.java

```diff
@@ -43,25 +39,29 @@
  * As the configuration is done via channel properties, this converter is usable via static thing types only.
  *
  * @author Henning Sudbrock - initial contribution
+ * @author Thomas Weißschuh - support for attribute-based buttons
  */
-public class ZigBeeConverterGenericButton extends ZigBeeBaseChannelConverter implements ZclCommandListener {
+public class ZigBeeConverterGenericButton extends ZigBeeBaseChannelConverter implements ZclCommandListener, ZclAttributeListener {
 
-    private Logger logger = LoggerFactory.getLogger(this.getClass());
+    private Logger logger = LoggerFactory.getLogger(ZigBeeConverterGenericButton.class);
```

Comment:
I suggest to leave the logger definition as it was (i.e., using `this.getClass()`); This will yield the correct logger names when using subclassing. Although subclassing is not used here, using `this.getClass()` frees the reader from having to think about why the class object is used instead of the more dynamic `this.getClass()`.

Reply:
I would like to learn here: What is the advantage of not using `this.getClass()`, but rather using `ZigBeeConverterGenericButton.class`?

Reply:
When looking for the source (code) of an issue one knows where to find it.
Also, it would be more predictable which logger to configure when the logger name does not depend on a (potentially unknown) subclass.

Reply:
Thanks for your feedback!